src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -0,0 +1,322 @@
+    if (
+      FocusTrapZone._focusStack.length &&
+      this === FocusTrapZone._focusStack[FocusTrapZone._focusStack.length - 1]
+    ) {
+      const clickedElement = ev.target as HTMLElement
+
+      if (clickedElement && !this._root.current.contains(clickedElement)) {
+        this.focus()
+        ev.preventDefault()
+        ev.stopPropagation()
+      }
+    }
+  }
+
+  private _subscribeToEvents = () => {
+    const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
+    if (forceFocusInsideTrap) {
+      eventStack.sub('focus', this._forceFocusInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+
+    if (!isClickableOutsideFocusTrap) {
+      eventStack.sub('click', this._forceClickInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+  }
+
+  private _unsubscribeFromEvents = () => {
+    const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
+    if (forceFocusInsideTrap) {
+      eventStack.unsub('focus', this._forceFocusInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+
+    if (!isClickableOutsideFocusTrap) {
+      eventStack.unsub('click', this._forceClickInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+  }
+
+  private _getPreviouslyFocusedElementOutsideTrapZone = () => {
+    const { elementToFocusOnDismiss } = this.props
+    let previouslyFocusedElement = this._previouslyFocusedElementOutsideTrapZone
+
+    if (elementToFocusOnDismiss && previouslyFocusedElement !== elementToFocusOnDismiss) {
+      previouslyFocusedElement = elementToFocusOnDismiss
+    } else if (!previouslyFocusedElement) {
+      previouslyFocusedElement = document.activeElement as HTMLElement
+    }
+
+    return previouslyFocusedElement
+  }
+
+  private _hideContentFromAccessibilityTree = () => {
+    const elements = (document.body && document.body.children) || []
+
+    if (elements.length && !document.body.contains(this._root.current)) {
+      // In case popup render options will change
+      throw new Error(
+        'Body does not contain trap zone element as expected. If it is done intentionally, please, make sure to update FocusTrapZone.',
+      )
+    }
+
+    // loop through all body's children, except the last one - which is the popup
+    for (let index = 0; index < elements.length - 1; index++) {
+      const element = elements[index]
+
+      if (
+        element.nodeName !== 'SCRIPT' &&
+        element.nodeName !== 'STYLE' &&
+        element.getAttribute('aria-hidden') !== 'true'
+      ) {
+        element.setAttribute('aria-hidden', 'true')
+        element.setAttribute(HIDDEN_FROM_ACC_TREE, 'true')
+      }
+    }
+  }
+
+  private _showContentInAccessibilityTree = () => {
+    const hiddenElements = document.querySelectorAll(`[${HIDDEN_FROM_ACC_TREE}="true"]`)
+    for (let index = 0; index < hiddenElements.length; index++) {
```

Comment:
nit: `hiddenElements.forEach((hiddenElement, index) => ...)`

Reply:
yes, plus for loop is faster then forEach

Reply:
cannot take `faster` argument when we are using `{...obj, ...anotherObj }` expressions everywhere - and those introduce extremely low overhead in contrast to actual rendering or even requests to DOM elements (like `query` ones) :) At the same time, correctness point is very appealing, thanks!